## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#921ae7d0-e3c4-420c-bc94-63e2450aadfa) |
| mint-large-bucket.sh | :x: [more...](#637d50c0-eb9b-48ea-86df-8b17497cb016) |
| mint-compression-dist-xl.sh | :x: [more...](#501f1316-d8fc-4490-869d-dfdeccbd7f23) |
<a name="637d50c0-eb9b-48ea-86df-8b17497cb016"></a>
### 8472-95a7d3f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31655
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 826e1aae31db:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 6 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 21 seconds
(5/14) Running awscli tests ... done in 2 minutes and 21 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 6 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 14 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 2 seconds
{
  "args": {},
  "duration": 96781,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="501f1316-d8fc-4490-869d-dfdeccbd7f23"></a>
### 8472-95a7d3f/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:32594
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 07be0304230f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 46 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 10 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 32 seconds
(8/14) Running minio-dotnet tests ... done in 42 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 15 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 46 seconds
(11/14) Running minio-js tests ... done in 55 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 19 seconds
{
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "args": {
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>e10ddd12-e7f7-41e2-b0cf-22316b892787</Key><BucketName>minio-py-test-db09552f-9a76-4a85-ab91-4b9bf0157e34</BucketName><Resource>/minio-py-test-db09552f-9a76-4a85-ab91-4b9bf0157e34/e10ddd12-e7f7-41e2-b0cf-22316b892787</Resource><RequestId>15D2AD36E97ED8C3</RequestId><HostId>96cb466f-b36f-4ced-853a-0a894ed88588</HostId></Error>",
    "response.status": 403,
    "bucket_name": "minio-py-test-db09552f-9a76-4a85-ab91-4b9bf0157e34",
    "object_name": "e10ddd12-e7f7-41e2-b0cf-22316b892787",
    "response.reason": "Forbidden",
    "presigned_get_object_url": "http://72.28.97.52:32594/minio-py-test-db09552f-9a76-4a85-ab91-4b9bf0157e34/e10ddd12-e7f7-41e2-b0cf-22316b892787?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191031%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191031T081418Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=67fbe6008b029b8756a6d30f06bd72dc1ab2fd6771332f769471d1a79ca3a96d",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-10-31T07-44-46Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15D2AD36E97ED8C3\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Thu, 31 Oct 2019 08:14:31 GMT\"]}}"
  },
  "status": "FAIL",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "duration": 111,
  "message": "AccessDenied: message: Access Denied"
}

Executed 11 out of 14 tests successfully.
```
<a name="921ae7d0-e3c4-420c-bc94-63e2450aadfa"></a>
### 8472-95a7d3f/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31469
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c72774f368a9:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-jcdhppnypzu52z",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365267,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D2ACF65C1B1101, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
Error: No such image: minio/minio:8472-95a7d3f

--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -252,12 +252,11 @@ func (fs *FSObjects) StorageInfo(ctx context.Context) StorageInfo {
 	if !fs.diskMount {
 		used = atomic.LoadUint64(&fs.totalUsed)
 	}
-	localPeer := GetLocalPeer(globalEndpoints)
 	storageInfo := StorageInfo{
 		Used:       []uint64{used},
 		Total:      []uint64{di.Total},
 		Available:  []uint64{di.Free},
-		MountPaths: []string{localPeer + fs.fsPath},
+		MountPaths: []string{fs.fsPath},
 	}
 	storageInfo.Backend.Type = BackendFS
 	return storageInfo

--- a/cmd/posix.go
+++ b/cmd/posix.go
@@ -321,10 +321,14 @@ type DiskInfo struct {
 // total free inodes and underlying filesystem.
 func (s *posix) DiskInfo() (info DiskInfo, err error) {
 	defer func() {
-		if err == errFaultyDisk {
+		if s != nil && err == errFaultyDisk {
 			atomic.AddInt32(&s.ioErrCount, 1)
 		}
 	}()
+
+	if s == nil {
+		return info, errFaultyDisk
+	}
 
 	if atomic.LoadInt32(&s.ioErrCount) > maxAllowedIOError {
 		return info, errFaultyDisk
@@ -344,7 +348,11 @@ func (s *posix) DiskInfo() (info DiskInfo, err error) {
 	if err != nil {
 		return info, err
 	}
-	localPeer := GetLocalPeer(globalEndpoints)
+
+	localPeer := ""
+	if globalIsDistXL {
+		localPeer = GetLocalPeer(globalEndpoints)
+	}
 
 	return DiskInfo{
 		Total:        di.Total,

--- a/cmd/xl-v1.go
+++ b/cmd/xl-v1.go
@@ -99,6 +99,10 @@ func getDisksInfo(disks []StorageAPI) (disksInfo []DiskInfo, onlineDisks, offlin
 
 	getPeerAddress := func(diskPath string) (string, error) {
 		hostPort := strings.Split(diskPath, SlashSeparator)[0]
+		// Host will be empty for xl/fs disk paths.
+		if hostPort == "" {
+			return "", nil
+		}
 		thisAddr, err := xnet.ParseHost(hostPort)
 		if err != nil {
 			return "", err
@@ -111,6 +115,7 @@ func getDisksInfo(disks []StorageAPI) (disksInfo []DiskInfo, onlineDisks, offlin
 	// Wait for the routines.
 	for i, err := range g.Wait() {
 		peerAddr, pErr := getPeerAddress(disksInfo[i].RelativePath)
+
 		if pErr != nil {
 			continue
 		}
@@ -122,6 +127,7 @@ func getDisksInfo(disks []StorageAPI) (disksInfo []DiskInfo, onlineDisks, offlin
 		}
 		if err != nil {
 			offlineDisks[peerAddr]++
+			continue
 		}
 		onlineDisks[peerAddr]++
 	}

--- a/go.mod
+++ b/go.mod
@@ -36,7 +36,6 @@ require (
 	github.com/klauspost/reedsolomon v1.9.3
 	github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
 	github.com/lib/pq v1.0.0
-	github.com/mattn/go-isatty v0.0.7
 	github.com/miekg/dns v1.1.8
 	github.com/minio/cli v1.22.0
 	github.com/minio/dsync/v2 v2.0.0

